### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Outkarm**
add all functions and files to the project

### CODE REVIEW

**Comment by brainskev** [comment]
## Status: Approved ✔️

**Hi @Outkarm,**

You did a really great job! 💯 💯💯💯
Congratulations! 🎉 🎉 🎉 🎉 

128382030-38e7de91-f6f2-45cf-b54e-8068e0446e4c

You may close this Pull Request and merge the changes
Keep being awesome dev's and good luck with your next milestone!

## To Highlight

✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally.
✅ Commit messages are meaningful.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding! 👏 👏 👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

File: README.md [hunk 1]
```diff
@@ -1 +1,78 @@
-# javascript-capstone-group-project
+![](https://img.shields.io/badge/Microverse-blueviolet)
```
Comment (1GORDON) [inline README.md:1]: Great job team 🤝

- [ ] As stated in the general requirements, kindly use the correct readme template and customize it accordingly. Here is the link to the correct [readme template](https://github.com/microverseinc/readme-template).

File: README.md [hunk 2]
```diff
@@ -1 +1,78 @@
...
+
+Give a ⭐ if you like this project!
+
+## Acknowledgments
+
+- Hat tip to anyone whose code was used
+- Hat tip to [Font Awesome](https://fontawesome.com) for icons
+- Hat tip to [Google APIs](https://fonts.googleapis.com) for fonts
+
+## 📝 License
```
Comment (1GORDON) [inline README.md:76]: Great job team 🤝

- [ ] [Optional] Kindly fix the broken MIT license link to make your readme file more professional.

File: src/modules/commentCounter.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { getComments } from './fetchcomment.js';
+
+const commentNum = async () => {
```
Comment (1GORDON) [inline src/modules/commentCounter.js:3]: Great job team 🤝

- [ ] [Optional] You may kindly make sure the comments counter updates in real-time. In this case, the comments-counter updates when the page reloads.

File: src/modules/commentCounter.test.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+describe('counting test', () => {
```
Comment (1GORDON) [inline src/modules/commentCounter.test.js:1]: Great job team 🤝

- [ ] You may kindly fix the failing test cases. A the moment both test cases are failing.
test-fail

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @Outkarm & @brainskev ✋

Good job so far!

## Highlights!
- Followed Gitflow ✔️ 
- Good commit messages ✔️
- Linters are passing ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/commentCounter.test.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  test('num the li', () => {
+    const meal = [];
+    const theCount = meal.length;
+    expect(theCount).toBe(0);
+  });
+
+  test('number the li', () => {
+    const meal = [{ rice: true }, { oat: false }];
+    const theCount = meal.length;
+    expect(theCount).toBe(2);
```
Comment (andersonlebon) [inline src/modules/commentCounter.test.js:11]: - [ ] Kindly note that you should write unit tests for some of the functionality of your app. I am really sorry but your code above is not testing any of the functionality of the App. and you are still having some failing tests, Kindly solve the as said by the first Reviewer

image

File: src/modules/itemCounter.test.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+    const theCount = meal.length;
+    expect(theCount).toBe(0);
+  });
+
+  test('number the meal', () => {
+    const meal = [{ rice: true }, { oat: false }];
+    const theCount = meal.length;
+    expect(theCount).toBe(2);
+  });
+});
```
Comment (andersonlebon) [inline src/modules/itemCounter.test.js:13]: - [ ] Kindly note that you should write unit tests for some of the functionality of your app. I am really sorry but your code above is not testing any of the functionality of the App. and you are still having some failing tests, Kindly solve the as said by the first Reviewer

image

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @Outkarm && @brainskev 

Great work on making the changes 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

almost-there

## Suggested changes
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question `@andersonlebon` so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by Lameck1 — CHANGES_REQUESTED** [review]
Hi @Outkarm, 

This is a manual review.

You have refactored your code as directed in the previous reviews. However, tests are still not running as expected 👇🏼:

image

This is because of this error 👇🏼:

image

There is a module missing in your project dependencies. Tests might be running successfully on your because this missing module might have been installed globally on your computer.

Cheers, and happy coding.

**Review by Lameck1 — CHANGES_REQUESTED** [review]
This an additional comment.

After installing the missing dependency with 👇🏼:

```bash
npm install --save-dev @babel/preset-typescript
```

The test now can run but all of the still fail as show below:

image

**Comment by Outkarm** [comment]
@Lameck1  please I have installed babel/preset typescript, please check

**Comment by Outkarm** [comment]
Capture

**Comment by Outkarm** [comment]
@Lameck1 please my life depends on this please sir approve

**Review by Lameck1 — APPROVED** [review]
Hi, @Outkarm,

This is a manual review.

Based on the information you've shared, tasks for this project are now considered successfully implemented 💯.

All the best with the next project.

You are free to merge this branch :shipit:
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

**Pull request merged**